server/graphql/types/src/types/repack.rs

```diff
@@ -0,0 +1,74 @@
+use async_graphql::*;
+use chrono::NaiveDateTime;
+use service::repack::query::Repack;
+
+use super::StockLineNode;
+
+pub struct RepackNode {
+    // Invoice id
+    pub id: String,
+    pub repack_id: String,
+    pub batch: Option<String>,
+    pub datetime: NaiveDateTime,
+    pub from: RepackStockLineNode,
+    pub to: RepackStockLineNode,
+}
+
+pub struct RepackStockLineNode {
+    pub stock_line: StockLineNode,
```

Comment:
Can get the packSize, numberOfPacks and Location from the from node, so I didn't add them in here 🤔

Reply:
yeah, I wondered about the choice of which fields to lift out - here and in the `RepackNode`
I agree tho - the properties are available, so there's no need to lift them up at this stage

Reply:
Location maybe wrong (it shows current location of stock line not the location of repack), we need the invoice line location

And I think we should be doing `reshaping` here rather then in front end (easier to test and reduces front end logic, which should ideally just be worried about presentation layer).

By reshaping i mean (below is what's displayed as per mock ups):

`Number of Pack`: to.packSize * to.numberOfPacks / from.nuberOfPacks (btw currently this logic is only in back end, with this RepackNode shape it would now be in both front end and back end).

Reply:
The `to` StockLine has the `Number of Packs` = input.num_packs * from.packSize / newPackSize... which is what is shown on central... your calculation shows the packSize not the number of packs? 🤔 

and the new location in the `to` StockLine is the new location (if any)?

Reply:
sorry i rushed my comment, yes denominator should be from.packSize

Reply:
> Is it supposed to be the Number of packs from insert input?

Yep

Reply:
> and the new location in the to StockLine is the new location (if any)? Location in the invoice_line for to StockLine is new_stock_line.location_id...

Hmm not sure if we are on the same page, but stock_line.location_id may not be invoice_line.location_id, since it could have moved.

Reply:
Sorry, so the location is supposed to be where the repack has moved to? Which means the `to` StockLine has that? 🤔